File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size())
            return false;

        unordered_map<char, int> hashMap{};
        for (int i = 0; i < s.size(); i++)
            hashMap[s[i]]++;

        for (int i = 0; i < t.size(); i++)
            hashMap[t[i]]--;

        for (int i = 0; i < s.size(); i++)
            if (hashMap[s[i]])
                return false;
        return true;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashMap = {} # number -> freq
        for num in nums:
            hashMap[num] = 1 + hashMap.get(num, 0)

        arr = [] # ([freq, number])
        for num,freq in hashMap.items():
            arr.append([freq,num])
        
        arr.sort() # sort based on freq

        res = []
        for i in range(k):
            res.append(arr.pop()[1]) # we want num, not freq

        return res